### DISCUSSION TITLE
Team 4061 Task Reports

### DISCUSSION THREAD

**Discussion by chauser** [wpilibsuite/2025Beta#47] [Task Reports]
### Task 1: Software Installation
Installed on Windows 11 (All Users) and Linux Ubuntu 22.04.
Only issue was: 

1. an "unknown publisher" pop-up on Windows when running WPILibInstaller, requiring clicking "more Information" and "Yes" to allow it to run (this is documented in the 2024 docs. I just don't like training people to bypass security measures by requiring them to do it.)

**Comment by sciencewhiz** [comment]
In the past, a signed executables could bypass that, which costs several hundred a year for a signing certificate. However, it appears now that only reduces the thresholds before a new binary doesn't get flagged. So I don't know any way around that.

**Comment by chauser** [comment]
Task 0
On a RoboRIO 1 the 2024 code on Java 2024.3.2 WPILib reports 21MB of free RAM and 124MB of free disk.
On a RoboRIO 1 this code has previously shown the behavior that the DS Restart Robot Code button does not work unless the NI Web server is disabled (and we have previously discussed the likelihood that this is due to lack of free RAM).

**Reply by chauser** [reply]
Task 3 Port Existing Robot Code to New System

The new vendordep plugin was helpful.
The change log mentions "Add ArmFeedforward calculate() overload that takes current and next velocity instead of acceleration". This applies to SimpleMotorFeedforward as well (and, I expect, ElevatorFeedforward). This could be a sub-bullet of the item "Discretize SimpleMotorFeedForward, ArmFeedForward and ElevatorFeedForward"

The ported code does not run on the RoboRio 1 I have available for testing. It gets an out-of-memory error after ~45 seconds or immediately when enabling. This is not too surprising given what I reported above for Task 0. I will test on a RoboRio 2 when I am able. 

Overall, the porting this year seemed more difficult and time-consuming than in the recent past. The Units-library conversion needed extensive changes (partly due to importer issues that I believe have now been fixed). For REV motor controller configuration, one needs to learn a quite different mental model of how it works and I'm not sure I still have it completely down. I think the new approach is good in that it encourages all the configuration to take place in one interaction with the motor controller hardware; I'm not so convinced that handling configuration of the integrated encoders and PID controllers in that same interaction is necessary or desirable. (That is, maybe naively, I would prefer to have separate `configure` methods for the encoders and pid controllers instead of embedding configurations for those in the configuration object that controls motor behavior -- inverted, idleMode, current limits, etc.)

More when I can try the code on a RoboRio 2.

**Reply by TheComputer314** [reply]
For the OOM crashes, can you put a link to your code in https://github.com/wpilibsuite/2025Beta/issues/64? Multiple users have been reporting OOM issues with RIO1s with 2025, but we currently don't have specific codebases that exhibit this behavior.
